src/internal.rs

```diff
@@ -255,7 +261,7 @@ pub struct Rav1dContext {
     pub(crate) inloop_filters: Rav1dInloopFilterType,
     pub(crate) decode_frame_type: Rav1dDecodeFrameType,
     pub(crate) drain: c_int,
-    pub(crate) frame_flags: PictureFlags,
+    pub(crate) frame_flags: RwLock<PictureFlags>,
```

Comment:
This can be an atomic instead I think.

If you do `cargo add atomig --features derive` and

```rust
#[derive(Clone, Copy, PartialEq, Eq, Hash, Default, Atom, AtomLogic)]
pub struct PictureFlags(u8);

bitflags! {
    impl PictureFlags: u8 {
        const NEW_SEQUENCE = 1 << 0;
        const NEW_OP_PARAMS_INFO = 1 << 1;
        const NEW_TEMPORAL_UNIT = 1 << 2;
    }
}
```

then you can do `Atomic<PictureFlags>`.

Reply:
See #679.